src/sympc/protocol/aby3/aby3.py

```diff
@@ -277,22 +294,133 @@ def bit_injection(x: MPCTensor, session: Session, ring_size: int) -> MPCTensor:
 
         args = [[share, str(ring_size)] for share in x.share_ptrs]
 
-        decompose = parallel_execution(ABY3.local_decomposition, session.parties)(args)
+        decomposed_shares = parallel_execution(
+            ABY3.local_decomposition, session.parties
+        )(args)
 
         # Using zip for grouping on pointers is compute intensive.
-        x1_sh = []
-        x2_sh = []
-        x3_sh = []
+        x1_share = []
+        x2_share = []
+        x3_share = []
 
-        for sh in decompose:
-            x1_sh.append(sh[0].resolve_pointer_type())
-            x2_sh.append(sh[1].resolve_pointer_type())
-            x3_sh.append(sh[2].resolve_pointer_type())
+        for share in list(
+            map(lambda x: x[0].resolve_pointer_type(), decomposed_shares)
+        ):
+            x1_share.append(share[0].resolve_pointer_type())
+            x2_share.append(share[1].resolve_pointer_type())
+            x3_share.append(share[2].resolve_pointer_type())
 
-        x1 = MPCTensor(shares=x1_sh, session=session, shape=x.shape)
-        x2 = MPCTensor(shares=x2_sh, session=session, shape=x.shape)
-        x3 = MPCTensor(shares=x3_sh, session=session, shape=x.shape)
+        x1 = MPCTensor(shares=x1_share, session=session, shape=x.shape)
+        x2 = MPCTensor(shares=x2_share, session=session, shape=x.shape)
+        x3 = MPCTensor(shares=x3_share, session=session, shape=x.shape)
 
         arith_share = x1 ^ x2 ^ x3
 
         return arith_share
+
+    @staticmethod
+    def full_adder(
+        a: List[MPCTensor], b: List[MPCTensor], session: Session
+    ) -> List[MPCTensor]:
+        """Perform bit addition on MPCTensors using a full adder.
+
+        Args:
+            a (List[MPCTensor]): MPCTensor with shares of bit.
+            b (List[MPCTensor]): MPCTensor with shares of bit.
+            session (Session): session the share belongs to.
+
+        Returns:
+            result (List[MPCTensor]): Result of the operation.
+
+        TODO: Should modify ripple carry adder to parallel prefix adder,currently unused.
+        """
+        ring_size = session.ring_size
+        ring_bits = get_nr_bits(ring_size)
+        c = [0 for _ in range(ring_bits + 1)]  # carry bits of addition.
```

Comment:
MAYBE: Can't `c` be only a value that has initial value 0?

Reply:
You mean like `c=0` , we use it as a list in next step george,couldn't get you clearly.
https://github.com/OpenMined/SyMPC/blob/db57826b8a842dd24d93d241bf745d3fcaeaf3e7/src/sympc/protocol/aby3/aby3.py#L343

Reply:
Something like this:
```
c = 0
result: List[MPCTensor] = []
        for idx in range(ring_bits):
            s = a[idx] + b[idx] + c
            c = a[idx] * b[idx] + c * (a[idx] + b[idx])
            result.append(s)
```

Reply:
Ahh Nice ,I would change it.

Reply:
Done